graphene/src/document.rs

```diff
@@ -4,15 +4,17 @@ use std::{
 };
 
 use glam::{DAffine2, DVec2};
+use serde::{Deserialize, Serialize};
 
 use crate::{
 	layers::{self, Folder, Layer, LayerData, LayerDataType, Shape},
 	DocumentError, DocumentResponse, LayerId, Operation, Quad,
 };
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Deserialize, Serialize)]
 pub struct Document {
 	pub root: Layer,
+	#[serde(skip)]
```

Comment:
we could think about including the current document hash here (hasher.finish()) but I am not sure if that would even be useful